schema/deploy/function_set_random_id.sql

```diff
@@ -0,0 +1,37 @@
+-- Deploy ggircs-portal:function_set_random_id to pg
+-- requires: schema_ggircs_portal
+
+begin;
+create extension if not exists pgcrypto;
+create function ggircs_portal.set_random_id()
+  returns trigger as $$
+
+declare
+  new_id varchar(1000);
+begin
+  while new_id in (select id from ggircs_portal.certification_url) or new_id is null loop
+    select gen_random_bytes(64) into new_id;
```

Comment:
Thinking about this a bit more, we need to encode the `bytea` result from `gen_random_bytes` into a URL-safe id to be stored as `varchar(1000)`.

```suggestion
    -- Securely generate a pseudo-random hash and re-encode it as a base64 string.
    new_id := encode(gen_random_bytes(64), 'base64');

    -- Base64 encoding contains 2 URL unsafe characters by default.
    -- The URL-safe version has these replacements.
    new_id := replace(key, '/', '_'); -- url safe replacement
    new_id := replace(key, '+', '-'); -- url safe replacement
```

Reply:
I added this in to the deploy file, but replacing the '+' with gen_random_bytes(64) wasn't working. There was always a '+' generated in the id, but not picked up on the front end. Maybe a '+' is automatically added to the string at a certain length? To resolve this for now I dropped the bytes to 32.

Reply:
nevermind it works.